### PR TITLE
For #2834: Delete all Private Tabs redesign

### PR DESCRIPTION
**Pull request by jyeontaek**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by jyeontaek** [comment]
@sblatz Did we want to add tests for this feature as well?

File: app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt [hunk 1]
```diff
@@ -53,6 +56,29 @@ class ToolbarIntegration(
                     return@run
                 }
 
+                if (isPrivate) {
+                    val deleteIcon = context.getDrawable(R.drawable.ic_delete)
+                    deleteIcon?.setColorFilter(
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt]: Is the color filter needed? The fill is already set to primaryText.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt [hunk 2]
```diff
@@ -53,6 +56,29 @@ class ToolbarIntegration(
...
+                            context,
+                            ThemeManager.resolveAttribute(R.attr.primaryText, context)
+                        ), PorterDuff.Mode.SRC_IN
+                    )
+                    deleteIcon?.let {
+                        val deleteSessions = BrowserToolbar.Button(
+                            deleteIcon,
+                            context.getString(R.string.private_browsing_delete_session),
+                            listener = {
+                                Navigation.findNavController(toolbar)
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt]: nit: use `toolbar.findNavController()` extension function

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 3]
```diff
@@ -305,6 +305,16 @@ class HomeFragment : Fragment(), AccountObserver {
         (activity as AppCompatActivity).supportActionBar?.hide()
 
         requireComponents.backgroundServices.accountManager.register(this, owner = this)
+
+        val closeAllPrivateTabs = arguments
+            ?.let(HomeFragmentArgs.Companion::fromBundle)
+            ?.let { it.closeAllPrivateTabs } ?: false
+
+        arguments = null
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: Do the arguments need to be cleared out? Other parts of the fragment could use them.

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 4]
```diff
@@ -305,6 +305,16 @@ class HomeFragment : Fragment(), AccountObserver {
...
+
+        if (closeAllPrivateTabs) {
+            handleTabAction(TabAction.CloseAll(true))
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: It feels like a bit of a hack to use navigation arguments to call this function. Could a `deleteAllPrivateBrowsingTabs` function be passed directly to ToolbarIntegration?

Comment (jyeontaek) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt]: Thanks for the review! I'll try to come up with a better solution.

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/TabHeaderViewHolder.kt [hunk 5]
```diff
@@ -63,7 +75,9 @@ class TabHeaderViewHolder(
         val headerTextResourceId =
             if (isPrivate) R.string.tabs_header_private_title else R.string.tab_header_label
         view.header_text.text = view.context.getString(headerTextResourceId)
-        view.tabs_overflow_button.visibility = if (hasTabs) View.VISIBLE else View.GONE
+        view.share_tabs_button.visibility = if (isPrivate && hasTabs) View.VISIBLE else View.GONE
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/TabHeaderViewHolder.kt]: nit: use the `isVisible` extension function

**Comment by sblatz** [comment]
@jyeontaek just some linting errors to fix too: 
image

**Pull request merged**